lib/appmap/rails/sql_handler.rb

```diff
@@ -71,21 +71,32 @@ def execute_query(sql)
           end
         end
 
-        class ActiveRecordExaminer
-          def server_version
-            ActiveRecord::Base.connection.try(:database_version) ||\
-              warn("Unable to determine database version for #{database_type.inspect}")
-          end
-
-          def database_type
-            type = ActiveRecord::Base.connection.adapter_name.downcase.to_sym
-            type = :postgres if type == :postgresql
+        if defined?(ActiveRecord)
```

Comment:
We are only using this class if ActiveRecord is defined:

https://github.com/applandinc/appmap-ruby/blob/42917c73c7a2459caee420072544376bd26c76b2/lib/appmap/rails/sql_handler.rb#L54

So this line seems redundant with that? Could anything go wrong if this file is loaded before ActiveRecord is required?

Reply:
Yeah, this is no good, either.